File: cajero_logica.js


let dinero = [10, 20, 50, 100]
let valor = 250
let retiro = []
let total = 0
let carry = 0

while (total !== valor){

    for (let i=carry; i < dinero.length; i++){
        if (dinero[i] + total <= valor){
            console.log("total es menor a valor todavia");
            
            retiro.push(dinero[i])
            total += dinero[i]
            console.log(dinero[i]);
            
        }
    }
    
    carry = carry > 4 ? 0 : carry + 1
    
}

console.log(retiro);
console.log("total: "+ total);

File: cajero_logica.py

dinero = [10, 20, 50, 100]
retiro = []
valor = 200 # valor que se quiere retirar, siempre multiplos de 10
total = 0

i=0

while total != valor:

    if(dinero[i] + total <= valor):
        retiro.append(dinero[i])
        total += dinero[i]
        i += 1
    else:
        i -= 1

    if(total != valor and i == 4):
        i -= 1

    

print(f"Los billetes que les fueron dados: {retiro}")
print(f"Monto total: {total}")

